pkg/virt-operator/resource/apply/fuzz_test.go

```diff
@@ -0,0 +1,362 @@
+			stores.SCCCache = informers.SCC.GetStore()
+
+			var b bytes.Buffer
+			writer := bufio.NewWriter(&b)
+			for _ = range 2 {
+				// Create a random SecurityContextConstraints
+				// for the manifest.
+				scc1 := &secv1.SecurityContextConstraints{}
+				fdp.Fuzz(scc1)
+				marshalutil.MarshallObject(scc1, writer)
+				// Split the resources in the manifest
+				writer.WriteString("---")
+
+				// Create a random SecurityContextConstraints
+				// and add it to the cache.
+				scc2 := &secv1.SecurityContextConstraints{}
+				fdp.Fuzz(scc2)
+				stores.SCCCache.Add(scc2)
+			}
+			writer.Flush()
+			resourceBytes = b.Bytes()
+		case 6:
+			// prepare for testing createOrUpdateValidatingAdmissionPolicyBindings
+			// The preparations are:
+			// - Create manifest of 2 random ValidatingAdmissionPolicyBinding
+			// - Add 2 random ValidatingAdmissionPolicyBinding to the cache.
+			var informers util.Informers
+			informers.ValidationWebhook, _ = testutils.NewFakeInformerFor(&admissionregistrationv1.ValidatingAdmissionPolicyBinding{})
+			stores.ValidationWebhookCache = informers.ValidationWebhook.GetStore()
+			var b bytes.Buffer
+			writer := bufio.NewWriter(&b)
+			for _ = range 2 {
+				// Create a random ValidatingAdmissionPolicyBinding
+				// for the manifest
+				validatingAdmissionPolicyBinding1 := &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}
+				fdp.Fuzz(validatingAdmissionPolicyBinding1)
+				marshalutil.MarshallObject(validatingAdmissionPolicyBinding1, writer)
+				// Split the resources in the manifest
+				writer.WriteString("---")
+
+				// Create a random ValidatingAdmissionPolicyBinding
+				// and add it to the cache.
+				validatingAdmissionPolicyBinding2 := &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}
+				fdp.Fuzz(validatingAdmissionPolicyBinding2)
+				stores.ValidationWebhookCache.Add(validatingAdmissionPolicyBinding2)
+			}
+			writer.Flush()
+			resourceBytes = b.Bytes()
+		default:
+			return
+		}
+
+		// Create the reconciler
+		// At this point, the fuzzer has prepared the specific resources
+		// for the target API. It can now create the the reconciler
+		// and the caches and mocking it needs.
+		ctrl := gomock.NewController(t)
+		clientset := kubecli.NewMockKubevirtClient(ctrl)
+		kv := &v1.KubeVirt{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: Namespace,
+			},
+		}
+		expectations := &util.Expectations{}
+		expectations.DaemonSet = controller.NewUIDTrackingControllerExpectations(controller.NewControllerExpectationsWithName("DaemonSet"))
+		expectations.PodDisruptionBudget = controller.NewUIDTrackingControllerExpectations(controller.NewControllerExpectationsWithName("PodDisruptionBudgets"))
+
+		mockPodCacheStore := &cache.FakeCustomStore{}
+
+		stores.InfrastructurePodCache = mockPodCacheStore
+		stores.InstallStrategyConfigMapCache = cache.NewStore(cache.MetaNamespaceKeyFunc)
+
+		pdbClient := fake.NewSimpleClientset()
+		dsClient := fake.NewSimpleClientset()
```

Comment:
Deprecated

Reply:
Could you elaborate? I see this used in other tests like here: https://github.com/kubevirt/kubevirt/blob/bd0ed684acc4df32033b549aa179cc6b3bd7bdc8/pkg/virt-operator/resource/apply/apps_test.go#L87

Reply:
Indeed, it seems like we have yet to switch to an alternative function, all good!